Penteract/Assets/Scripts/AIDuke.cpp

```diff
@@ -216,7 +218,11 @@ void AIDuke::Update() {
 			troopsCounter -= 0.0063;
 			float3 dir = player->GetComponent<ComponentTransform>()->GetGlobalPosition() - ownerTransform->GetGlobalPosition();
 			movementScript->Orientate(dir);
-			dukeCharacter.ThrowBarrels();
+			throwBarrelTimer -= Time::GetDeltaTime();
+			if (throwBarrelTimer <= 0) {
+				dukeCharacter.ThrowBarrels();
+				throwBarrelTimer = 5;
```

Comment:
You should have a currentBarrelTimer and a BarrelTimer, the BarrelTimer is the value you set from inspector, then you should decrease the currentBarrelTimer. And when it is 0 you set the currentBarrelTimer to BarrelTimer